### PR TITLE
New type bounds

### PR DESCRIPTION
**Pull request by DarkWingMcQuack**
A refactoring of the typebounds for the effect type `F`.
Feel free to throw this pull request away, i am not an expert on this

### CODE REVIEW

File: src/main/scala/io/github/paoloboni/http/HttpClient.scala [hunk 1]
```diff
@@ -100,8 +97,8 @@ sealed class HttpClient[F[_]: Async: Client: LogWriter](implicit
         }
       }
       handled <- decoded.fold(
-        decodingFailure => E.raiseError(decodingFailure),
-        response => F.pure(response)
+        decodingFailure => implicitly[Async[F]].raiseError(decodingFailure),
```
Comment (paoloboni) [inline src/main/scala/io/github/paoloboni/http/HttpClient.scala:100]: Can you change this to
```suggestion
        decodingFailure => Async[F].raiseError(decodingFailure),
```

File: src/main/scala/io/github/paoloboni/http/HttpClient.scala [hunk 2]
```diff
@@ -100,8 +97,8 @@ sealed class HttpClient[F[_]: Async: Client: LogWriter](implicit
...
+        response => response.pure[F]
```
Comment (paoloboni) [inline src/main/scala/io/github/paoloboni/http/HttpClient.scala:101]: I like this syntax!

File: src/main/scala/io/github/paoloboni/http/ratelimit/RateLimiter.scala [hunk 3]
```diff
@@ -32,11 +33,11 @@ import io.github.paoloboni.binance.RateLimitType
 import scala.concurrent.duration.{DurationInt, FiniteDuration}
 
 object RateLimiter {
-  def make[F[_]: Temporal: Spawn](
+  def make[F[_]: Temporal](
```
Comment (paoloboni) [inline src/main/scala/io/github/paoloboni/http/ratelimit/RateLimiter.scala:36]: Spawn not explicitly required here 👍

**Review by paoloboni — APPROVED** [review]
Nice cleanup!

File: src/main/scala/io/github/paoloboni/http/HttpClient.scala (hunk 2 above)
Comment (DarkWingMcQuack) [reply src/main/scala/io/github/paoloboni/http/HttpClient.scala:101]: yeah i like the syntax of calling `.pure[F]` much more than calling `F.pure(...)`, this obviously works for all kinds of effects which is  quite handy

**Pull request merged**